### ISSUE TITLE
nodetool assassinate

### ISSUE THREAD

**Issue by tzach** [scylladb/scylla#435]
Forcefully removes a dead node without re-replicating any data. It is a last resort tool if you cannot successfully use `nodetool removenode` #243 
http://docs.datastax.com/en/cassandra/2.2/cassandra/tools/toolsAssassinate.html

**Comment by tzach** [comment]
This is a Cassandra 2.2 operation, previously supported via JMX
https://issues.apache.org/jira/browse/CASSANDRA-7935

**Comment by slivne** [comment]
@tzach this is not in the api workplan excel do we need this for GA ?

**Comment by glommer** [comment]
We have faced many situations in which we just have to force gossip to forget about the nodes.
As far as I understand, the API even exists in Scylla but it is hardly tested and @asias keeps recommending against it unless it is an absolute emergency.

Either through a better tested and more solid nodetool assassinate or other APIs, I would like to have more control about the gossip state of nodes allowing us to force removes, etc.

**Comment by tzach** [comment]
@vladzcloudius is this still relevant?

**Comment by vladzcloudius** [comment]
> @vladzcloudius is this still relevant?

@kostja @kbr- Do we have alternatives to the above today? 
In gist: is there a way to forcefully remove a node without streaming using a single command?

**Comment by kostja** [comment]
* Vladislav Zolotarov ***@***.***> [24/08/30 08:26]:
> > @vladzcloudius is this still relevant?
> 
> @kostja @kbr- Do we have alternatives to the above today? 
> In gist: is there a way to forcefully remove a node without streaming using a single command?

No, it is not implemented.

-- 
Konstantin Osipov, Moscow, Russia

**Comment by vladzcloudius** [comment]
> * Vladislav Zolotarov ***@***.***> [24/08/30 08:26]:
> > @vladzcloudius is this still relevant? @kostja @kbr- Do we have alternatives to the above today? In gist: is there a way to forcefully remove a node without streaming using a single command?
> No, it is not implemented.
> […](#)
> -- Konstantin Osipov, Moscow, Russia

Then probably we want to give it a thought. 
I'm referring a super intrusive mode of course - something only ScyllaDB employees or someone who really knows what he/she's doing should be using.

I'm thinking of a `nodetool` or probably a "scylla-tool" (from @denesb tool box) command.

I can also imagine that it may also have a non-trivial requirements, e.g. full outage and handling every node in a maintenance mode one by one.

However, obviously, the less restrictive it is - the better.

As we saw not once (@kbr-, @kostja and @bhalevy probably were the most involved representatives from Eng in those incidents) on very rare (but important) occasions you really need to have such a "hammer" to fix the cluster's Topology configuration when it gets out of whack...

**Comment by kostja** [comment]
* Vladislav Zolotarov ***@***.***> [24/08/31 12:02]:
> Then probably we want to give it a thought. 
> I'm referring a super intrusive mode of course - something only ScyllaDB employees or someone who really knows what he/she's doing should be using.
> 
> I'm thinking of a `nodetool` or probably a "scylla-tool" (from @denesb tool box) command.
> 
> I can also imagine that it may also have a non-trivial requirements, e.g. full outage and handling every node in a maintenance mode one by one.
> 
> However, obviously, the less restrictive it is - the better.
> 
> As we saw not once ***@***.***, @kostja and @bhalevy probably were the most involved representatives from Eng in those incidents) on very rare (but important) occasions you really need to have such a "hammer" to fix the cluster's Topology configuration when it gets out of whack... 

I agree it could be useful. 

@mykaul @kbr-scylla?

-- 
Konstantin Osipov, Moscow, Russia

**Comment by bhalevy** [comment]
I agree as well.  In the past we contemplated reviving `nodetool removenode --force` for a similar effect of forcibly removing a node and its token ownership from the cluster, without rebuilding its tokens using repair.
It's leaves the cluster in a degraded state, but I think it's akin to and not more dangerous than increasing RF by 1, with vnodes, before running repair, to rebuild the extra replica.

**Comment by bhalevy** [comment]
> I agree as well. In the past we contemplated reviving `nodetool removenode --force` for a similar effect of forcibly removing a node and its token ownership from the cluster, without rebuilding its tokens using repair. It's leaves the cluster in a degraded state, but I think it's akin to and not more dangerous than increasing RF by 1, with vnodes, before running repair, to rebuild the extra replica.

Conflicts with https://github.com/scylladb/scylladb/pull/15834

**Comment by kostja** [comment]
* Benny Halevy ***@***.***> [24/09/01 14:38]:
> > I agree as well. In the past we contemplated reviving `nodetool removenode --force` for a similar effect of forcibly removing a node and its token ownership from the cluster, without rebuilding its tokens using repair. It's leaves the cluster in a degraded state, but I think it's akin to and not more dangerous than increasing RF by 1, with vnodes, before running repair, to rebuild the extra replica.
> 
> Conflicts with https://github.com/scylladb/scylladb/pull/15834

I wouldn't say it's a conflict - we need to remove the old
implementation without question, it doesn't work.

-- 
Konstantin Osipov, Moscow, Russia

**Comment by bhalevy** [comment]
> * Benny Halevy ***@***.***> [24/09/01 14:38]:
> > I agree as well. In the past we contemplated reviving `nodetool removenode --force` for a similar effect of forcibly removing a node and its token ownership from the cluster, without rebuilding its tokens using repair. It's leaves the cluster in a degraded state, but I think it's akin to and not more dangerous than increasing RF by 1, with vnodes, before running repair, to rebuild the extra replica. Conflicts with #15834
> I wouldn't say it's a conflict - we need to remove the old implementation without question, it doesn't work.
> […](#)
> -- Konstantin Osipov, Moscow, Russia

or fix it.  aren't we committed to the published api?

**Comment by kbr-scylla** [comment]
What `nodetool removenode --force` does is undefined and the implementation does not make any sense to me.
